community/kernel/src/main/java/org/neo4j/kernel/api/KernelTransaction.java

```diff
@@ -96,6 +98,10 @@
      */
     Statement acquireStatement();
 
+    NodeCursor nodeCursor();
+
+    PropertyCursor propertyCursor();
```

Comment:
This itches a little... it feels like a hack to accommodate Core API. I guess it's safe as long as Core API transactions are thread bound and this cursors are not embedded in returned iterators or similar.

WDYT?